### PR TITLE
Adding the depends annotation to Integration Tests

### PR DESCRIPTION
**Pull request by eduard13**
## Purpose of this pull request

This pull request (PR) adds a new page with additional information related to **depends** annotation that is widely used on writing Integration Tests. 

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.1/test/integration/annotations/depends.html
- https://devdocs.magento.com/guides/v2.2/test/integration/annotations/depends.html
- https://devdocs.magento.com/guides/v2.3/test/integration/annotations/depends.html

## Links to Magento source code

- https://github.com/magento/magento2/blob/36097739bbb0b8939ad9a2a0dadee64318153dca/dev/tests/integration/testsuite/Magento/Catalog/Model/Layer/Filter/CategoryTest.php#L86
- https://github.com/magento/magento2/blob/36097739bbb0b8939ad9a2a0dadee64318153dca/dev/tests/integration/testsuite/Magento/Reports/Block/Adminhtml/Sales/Sales/GridTest.php#L49

whatsnew
Added new topic about using the [depends annotation](https://devdocs.magento.com/guides/v2.3/test/integration/annotations/depends.html) when writing integration tests.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by rogyar** [comment]
Hi @eduard13. That's a great topic. Thank you for the collaboration

File: guides/v2.1/test/integration/annotations/depends.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ * @depends methodName
+ */
+```
+
+{:.bs-callout .bs-callout-warning}
+Please note that dependencies don't define the order in which the test methods are executed, so you still have to keep to proper order for your methods. 
+
+### Example 1
+
+Lets check the following basic example.
```
Comment (erikmarr) [inline guides/v2.1/test/integration/annotations/depends.md]:
```suggestion
Let's check the following basic example.
```

File: guides/v2.1/test/integration/annotations/depends.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+public function testNumber($number)
+{
+    $this->assertEquals(2, $number);
+}
+```
+
+### Example 2
+
+{:.bs-callout .bs-callout-info}
+Using more than one dependencies, arguments will be passed in the annotations' defined order.
```
Comment (erikmarr) [inline guides/v2.1/test/integration/annotations/depends.md]:
```suggestion
If using multiple dependencies, arguments are passed in the annotations' defined order.
```

File: guides/v2.1/test/integration/annotations/depends.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+public function testNumber(int $one, int $two)
+{
+    $this->assertEquals(1, $one);
+    $this->assertEquals(2, $two);
+}
+```
+
+### Example 3
+
+Let's check the following practical example, where we'll be checking the customer email by customer id.
```
Comment (erikmarr) [inline guides/v2.1/test/integration/annotations/depends.md]:
```suggestion
Let's check the following practical example, where we'll be checking the customer email by customer ID.
```

File: guides/v2.1/test/integration/annotations/depends.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+ *
+ * @param CustomerInterface $customer
+ */
+public function testEmail(CustomerInterface $customer)
+{
+    $this->assertEquals('customer@example.com', $customer->getEmail());
+}
+```
+
+You can read more about PHPUnit Depends annotation [here](https://phpunit.readthedocs.io/en/8.2/annotations.html#depends){:target="_blank"}.
```
Comment (erikmarr) [inline guides/v2.1/test/integration/annotations/depends.md]:
```suggestion
You can read more about PHPUnit dependency annotation [here](https://phpunit.readthedocs.io/en/8.2/annotations.html#depends){:target="_blank"}.
```

**Comment by eduard13** [comment]
Hi @erikmarr, thank you for your suggestions, they were applied.

**Comment by erikmarr** [comment]
running tests

**Comment by erikmarr** [comment]
Thanks @eduard13!

**Pull request merged**